sspl_test/coverage/coverage_setup.py

```diff
@@ -0,0 +1,185 @@
+#! /usr/bin/python3
+
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+import shutil
+import os
+import pwd
+import signal
+import psutil
+import time
+import sys
+from cortx.utils.process import SimpleProcess
+# Add the top level directory to the sys.path to access classes
+topdir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+os.sys.path.insert(0, topdir)
+from framework.base.sspl_constants import DATA_PATH
+
+SSPL_LL_D = "/opt/seagate/cortx/sspl/low-level/sspl_ll_d"
+REPORT_PATH = f"{DATA_PATH}coverage/sspl_xml_coverage_report.xml"
+
+PATCH_1 = """\
+from coverage import Coverage
+co = Coverage(
+        data_file='%scoverage/.sspl_coverage_report',
+        include="/opt/seagate/*",
+        omit=['*/.local/*', '*/usr/*'],
+        # config_file='%scoverage/.coveragerc',
+        )\
+"""%(DATA_PATH, DATA_PATH)
+
+PATCH_2 = """\
+        logger.info("Starting coverage report scope")
+        co.start()\
+"""
+
+PATCH_3 = """\
+def generate_cov_report(signal_number, frame):
+    logger.info('Ending Coverage Scope')
+    co.stop()
+    logger.info('coverage object stopped.')
+    co.save()
+    logger.info('coverage info saved.')
+    cov_per = co.xml_report(ignore_errors=True,
+        outfile='%scoverage/sspl_xml_coverage_report.xml')
+    logger.info(f'XML coverage report generated with coverage of {cov_per} percentage.')
+    ## Enable below code to inable HTML report generation
+ 	# html_cov_per = co.html_report(
+ 	#                     directory='%scoverage/sspl_html_coverage',
+ 	#                     ignore_errors=True,
+ 	#                 )
+ 	# logger.info(f'HTML coverage report geverated with coverage of {html_cov_per} percentage.')\
+"""%(DATA_PATH, DATA_PATH)
+
+PATCH_4 = """\
+    signal.signal(signal.SIGUSR1, generate_cov_report)\
+"""
+
+def coverage_setup():
+    """Installs pip3 coverage package. Injects different patches code from
+       coverage_code file to sspl_ll_d file, also creates target directory
+       for code coverage report and assigns permission to the directory.
+    """
```

Comment:
Can add 1 blank line, but it does not follow the way we write a function description in sspl. as it would look like 
````
   """Installs pip3 coverage package. Injects different patches code from

      coverage_code file to sspl_ll_d file, also creates target directory
      for code coverage report and assigns permission to the directory.
   """